dockerfiles/synapse/start.sh

```diff
@@ -24,5 +24,4 @@ openssl x509 -req -in /conf/server.tls.csr \
   -CA /ca/ca.crt -CAkey /ca/ca.key -set_serial 1 \
   -out /conf/server.tls.crt
 
-exec python -m synapse.app.homeserver -c /conf/homeserver.yaml "$@"
-
+exec /synapse/.venv/bin/python -m synapse.app.homeserver -c /conf/homeserver.yaml "$@"
```

Comment:
similarly, change this to:

```suggestion
exec synapse_homeserver -c /conf/homeserver.yaml "$@"
```

... and then it becomes independent of the poetry change. It's a good thing to do anyway, if we're now telling people to use the entrypoint, per https://github.com/matrix-org/synapse/pull/12337/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR323.

Reply:
I'll do this separately.